lib/recursive_open_struct.rb

```diff
@@ -39,6 +44,17 @@ def [](name)
     send name
   end
 
+  def marshal_dump
+    [@table, @args]
```

Comment:
This seems like a reasonable set of variables to preserve. However, what would happen if the internal implementation changes? Should we be guarding against that by trying to version the marshalled format?